hostfactory/host_provider/src/cluster.py

```diff
@@ -18,31 +19,68 @@ class AutoscalingStrategy(enumerate):
     CAPACITY = "capacity"
     WEIGHTED = "weighted"
 
-# This is only for unit testing
-def allocation_strategy(node_mgr, template_id, slot_count, capacity_limit_timeout, vm_sizes):
-    per_vm_size = slot_count//len(vm_sizes)
-    remainder = slot_count % len(vm_sizes)
-    print(vm_sizes)
+def allocation_strategy(node_mgr, logger, template_id, slot_count, capacity_limit_timeout, vm_sizes, vm_dist):
+    result = None
+    check_allocate = None
+    logger.debug(f"vm dist {vm_dist}")
     for n,vmsize in enumerate(vm_sizes):
-        print(vmsize)
-        vm_slot_count = per_vm_size + (1 if n < remainder else 0)
-        print(f"Allocating {vm_slot_count} {vmsize}" )
+        vm_slot_count = vm_dist[vmsize]
+        logger.debug(f"Allocating {vm_slot_count} {vmsize}" )
         if vm_slot_count > 0:
-            node_mgr.allocate({"node.vm_size": vmsize, "weight": 1, "template_id": template_id, "capacity-failure-backoff": capacity_limit_timeout},
+            check_allocate=node_mgr.allocate({"node.vm_size": vmsize, "weight": 1, "template_id": template_id, "capacity-failure-backoff": capacity_limit_timeout},
                             slot_count=vm_slot_count,
                             allow_existing=False)
-        if vm_slot_count == 0:
-            print(f"Allocated required {slot_count}")
-            break
+            if check_allocate.status == "NoAllocationSelected":
+                logger.debug("No allocation selected")
```

Comment:
We can just log this instead
if check_allocate:
    logger.debug("0 new nodes allocated: {check_allocate.status}")
    
Otherwise you are hiding other failures